rust/src/helpers/table.rs

```diff
@@ -87,7 +87,11 @@ fn calculate_order(header_to_pos: &HashMap<String, usize>, ordering: &[&str]) ->
         .map(|(k, v)| (v, k * 2))
         .collect::<HashMap<&&str, usize>>();
 
-    let mut order: Vec<String> = header_to_pos.clone().into_keys().collect();
+    // order headers by pos
+    let mut header_pos: Vec<(String, usize)> = header_to_pos.clone().into_iter().collect();
+    header_pos.sort_by_key(|&(_, v)| v);
```

Comment:
Rather than sorting twice, can we just somehow append the position inside the store that happens immediately after this, that would be a better solution. Having two sorts make it look a bit suboptimal.